gio/src/subclass/list_model.rs

```diff
@@ -0,0 +1,55 @@
+use crate::auto::ListModel;
+use glib::subclass::prelude::*;
+use glib::translate::*;
+use glib::Cast;
+
+pub trait ListModelImpl: ObjectImpl {
+    fn get_item_type(&self, list_model: &Self::Type) -> glib::Type;
```

Comment:
```
 * The item type of a #GListModel can not change during the life of the
 * model.
```

Maybe something we want to enforce somehow? This could cause big problems otherwise :)

Reply:
I have no idea how I enforce such thing, any suggestions?

Reply:
You could store the first type returned in object qdata and check against that afterwards, and also make sure that all items ever returned are of that type or a subclass of it.

Reply:
I.e. check at runtime and panic otherwise.

The alternatively would be to make this a `pub trait ListModelImpl<T: StaticType + IsA<Object>>` or something scary like that. Not sure how much headache that causes when implementing it though. Maybe bad, maybe not at all :)

Reply:
I have tired the at runtime check, seems to compile fine ^^ I will try adding an example and see if it works as expected.  Although I had to enforce `glib::ObjectType` on whatever that implements the `ListModel` interface.

Reply:
> Although I had to enforce glib::ObjectType on whatever that implements the ListModel interface.

Yeah I think we're missing something somewhere to bind all those traits together.